File: examples/java/IsPrime.java
public class IsPrime {
    static boolean isPrime(int number) {
        if (number < 2) {
            return false;
        }

        for (int divisor = 2; divisor <= number / divisor; divisor++) {
            if (number % divisor == 0) {
                return false;
            }
        }

        return true;
    }

    static void check(boolean condition, String message) {
        if (!condition) {
            throw new AssertionError(message);
        }
    }

    public static void main(String[] args) {
        check(!isPrime(1), "1 should not be prime");
        check(isPrime(2), "2 should be prime");
        check(isPrime(17), "17 should be prime");
        check(!isPrime(21), "21 should not be prime");
        System.out.println("isPrime: ok");
    }
}

File: examples/c/is_prime.c
#include <assert.h>
#include <stdbool.h>
#include <stdio.h>

static bool is_prime(unsigned int number) {
    if (number < 2U) {
        return false;
    }

    for (unsigned int divisor = 2U; divisor <= number / divisor; divisor++) {
        if (number % divisor == 0U) {
            return false;
        }
    }

    return true;
}

int main(void) {
    assert(!is_prime(1U));
    assert(is_prime(2U));
    assert(is_prime(17U));
    assert(!is_prime(21U));
    printf("is_prime: ok\n");
    return 0;
}
